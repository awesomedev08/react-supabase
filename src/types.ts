import { PostgrestFilterBuilder } from '@supabase/postgrest-js'

export type Count = 'exact' | 'planned' | 'estimated'

export type Filter<Data> = (
    query: PostgrestFilterBuilder<Data>,
) => PostgrestFilterBuilder<Data>

export type PostgrestError = {
    message: string
    details: string
    hint: string
    code: string
}

export type Returning = 'minimal' | 'representation'
