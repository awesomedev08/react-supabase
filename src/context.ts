import { createContext, useContext } from 'react'
import { SupabaseClient } from '@supabase/supabase-js'

export const Context = createContext<SupabaseClient | undefined>(undefined)
export const Provider = Context.Provider
export const Consumer = Context.Consumer
Context.displayName = 'SupabaseContext'

export const useClient = (): SupabaseClient => {
    const client = useContext(Context)
    if (client === undefined)
        throw new Error('No client has been specified using Provider.')
    return client
}
