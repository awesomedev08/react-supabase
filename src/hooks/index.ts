export * from './auth'
export * from './data'
export * from './realtime'
export * from './use-client'
