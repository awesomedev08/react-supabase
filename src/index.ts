export { Consumer, Context, Provider, useClient } from './context'
export { useFilter, useSelect } from './hooks'
