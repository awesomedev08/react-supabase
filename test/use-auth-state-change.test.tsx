import { renderHook } from '@testing-library/react-hooks'

import { useAuthStateChange } from '../src'

describe('useAuthStateChange', () => {
    it('should throw when not inside Provider', () => {
        const { result } = renderHook(() => useAuthStateChange(jest.fn()))
        expect(() => result.current).toThrowErrorMatchingSnapshot()
    })
})
