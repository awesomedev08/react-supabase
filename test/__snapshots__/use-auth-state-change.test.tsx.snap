// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useAuthStateChange should throw when not inside Provider 1`] = `"No client has been specified using Provider."`;
